File: AndroidMVVMLogin/app/src/main/java/com/example/shakil/androidmvvmlogin/LoginViewModelFactory.java
package com.example.shakil.androidmvvmlogin;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;

import com.example.shakil.androidmvvmlogin.Interface.LoginResultCallbacks;

public class LoginViewModelFactory extends ViewModelProvider.NewInstanceFactory {
    private LoginResultCallbacks loginResultCallbacks;

    public LoginViewModelFactory(LoginResultCallbacks loginResultCallbacks) {
        this.loginResultCallbacks = loginResultCallbacks;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new LoginViewModel(loginResultCallbacks);
    }
}

File: KotlinMVVMDesign/app/src/main/java/com/example/shakil/kotlinmvvmdesign/LoginViewModelFactory.kt
package com.example.shakil.kotlinmvvmdesign

import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProvider
import com.example.shakil.kotlinmvvmdesign.Interface.LoginResultCallbacks

class LoginViewModelFactory(private val listener:LoginResultCallbacks): ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return LoginViewModel(listener) as T
    }
}